docs/protobuf_getting_started.md

```diff
@@ -0,0 +1,153 @@
+ * `-base_import_path` (as described above) specifies the import path that should be used in the generated protobufs. The path used in this example specifies the entire path from `$GOPATH/src`, since this will be the include path supplied to `protoc`.
+ * `-path` specifies the search path(s) that should be used to find dependencies of the input YANG modules. Multiple directories can be separated with a comma.
+ * `-output_dir` specifies the directory into which the output files for the schema should be written.
+ * `-package_name` (as described above) specifies the name of the top-level package that should be created for the output schema.
+ * `-enum_package` specifies the name that should be used for the package which stores global enumerated values. Such values are generated for:
+ 	* `identity` statements in the YANG schema.
+ 	* `typedef` statements which contain an `enumeration`.
+ * Finally, the `yang/rib/openconfig-rib-bgp.yang` argument specifies the YANG schema for which the protobufs should be generated.
+
+Running this command outputs the generated set of protobufs to the specified directory, in this case `demo/protobuf_getting_started/ribproto`.
+
+In order to generate code for these protobufs, we run the `protoc` compiler. When running `protoc` we must specify import paths that allow the include paths specified in the protobufs generated to be resolved. Since `-base_import_path` was specified to be the path to the GitHub `ygot` repo, we can simply set the import path to `$GOPATH/src`. 
+
+Additionally, since the generated protobufs depend upon the `yext.proto` and `ywrapper.proto` files, generated code is required for these files. The code for these files can be generated using:
+
+```
+go get -u github.com/google/protobuf  # Ensure referenced protobufs are downloaded
+cd $GOPATH/src/github.com/openconfig/ygot/proto/yext && go generate
+cd $GOPATH/src/github.com/openconfig/ygot/proto/ywrapper && go generate
+```
+
+Finally, to generate the code for the the protobufs, generated we can simply loop:
+
+```
+proto_imports=".:${GOPATH}/src/github.com/google/protobuf/src:${GOPATH}/src"
+find $GOPATH/src/github.com/openconfig/ygot/demo/protobuf_getting_started/ribproto -name "*.proto" | while read l; do
+  protoc -I=$proto_imports --go_out=. $l
+done
+```
+
+This leaves us with a generated `.pb.go` file for each `.proto` that was generated by `ygot`.
+
+## Using the Generated Protobufs in a Go program
+
+Since the generated set of Protobufs form a number of different packages, each of these Go packages needs to be imported within the calling application, as demonstrated in the `demo/protobuf_getting_started/demo.go` program. Once the relevant protobufs have been imported, the generated Protobuf structures can be used as per any other generated protobuf code.
+
+
```

Comment:
This is a really nice doc/example. I wish more library authors had high quality writeups like this.

Reply:
Hopefully it helps -- it's definitely been useful to have these docs elsewhere.